browser/brave_content_browser_client.cc

```diff
@@ -213,6 +218,31 @@ void BindCosmeticFiltersResources(
       std::move(receiver));
 }
 
+void BindBraveAdsHost(
+    content::RenderFrameHost* const frame_host,
+    mojo::PendingReceiver<brave_ads::mojom::BraveAdsHost> receiver) {
+  auto* context = frame_host->GetBrowserContext();
+  auto* profile = Profile::FromBrowserContext(context);
+
+  if (brave::IsRegularProfile(profile)) {
+    content::WebContents* web_contents =
```

Comment:
this is going to create a BraveAdsHost for every render frame, but we only bind the JS in the main frame https://github.com/brave/brave-core/pull/9611/files#diff-8bcf679ada73b96c236a1e379113804f357ebafdcd1a3be1ecd1a97d5277787dR49
One of these two things is wrong, but I'm not sure which

Reply:
It would create BraveAdsHost only in case of js api call. But call is possible for main frame only.
I actually followed pattern with Brave search api.

Reply:
The brave search api does not effectively convert a render frame to the owner webcontents and this is part of the reason why we shouldn't be passing the webcontents. There's no reason to pass in the webcontents so you should get the BrowserContext and pass that in instead